packages/lesswrong/server/callbacks/commentCallbacks.ts

```diff
@@ -128,7 +129,7 @@ getCollectionHooks("Comments").removeAsync.add(async function CommentsRemovePost
   const { postId } = comment;
 
   if (postId) {
-    const postComments = await Comments.find({postId}, {sort: {postedAt: -1}}).fetch();
+    const postComments = await Comments.find({postId, debateResponse: false, deleted: false, deletedPublic: false}, {sort: {postedAt: -1}}).fetch();
```

Comment:
I would guess you'd want

{deleted: false} OR {deleted: true, deletedPublic: false}

With the current version you can go ahead and get rid of deletedPublic, it's not doing anything extra.

I would also note that this is not null-safe, although my current view of the database shows that, while the column is nullable, none of the values are null.